src/rayleigh_quotient.jl

```diff
@@ -142,7 +142,7 @@ function _get_C_reduced(prob::ConstrainedRayleighQuotientProblem, alg::RQ_SPARSE
     inds_remove, inds_keep = _inds_to_remove_and_keep(b, length(b))
     bk = first(b[inds_remove]) # only one element, no support for b Span
     Ck = C[inds_remove, :] # replace mult by J with slicing, see above
```

Comment:
Careful with keeping sparsity in the line after this. I think we may need `Ck = transpose(C[only(inds_remove), :])` for `b*Ck` to be sparse.